pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,597 @@
+				// Otherwise, ignore updates to objects when metadata.Generation does not change.
+				return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+			},
+		}
+		err = c.Watch(&source.Kind{Type: t}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &operator.Installation{},
+		}, pred)
+		if err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch %s: %w", t, err)
+		}
+	}
+
+	// Watch for changes to KubeControllersConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.KubeControllersConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch KubeControllersConfiguration resource: %w", err)
+	}
+
+	// Watch for changes to FelixConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.FelixConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch FelixConfiguration resource: %w", err)
+	}
+
+	// Watch for changes to BGPConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.BGPConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch BGPConfiguration resource: %w", err)
+	}
+
+	if ri.enterpriseCRDsExist {
+		// Watch for changes to primary resource ManagementCluster
+		err = c.Watch(&source.Kind{Type: &operator.ManagementCluster{}}, &handler.EnqueueRequestForObject{})
+		if err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch primary resource: %v", err)
+		}
+
+		// Watch for changes to primary resource ManagementClusterConnection
+		err = c.Watch(&source.Kind{Type: &operator.ManagementClusterConnection{}}, &handler.EnqueueRequestForObject{})
+		if err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch primary resource: %v", err)
+		}
+
+		// Watch the internal manager TLS secret in the calico namespace, where it's copied for kube-controllers.
+		if err = utils.AddSecretsWatch(c, render.ManagerInternalTLSSecretName, common.CalicoNamespace); err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch secret '%s' in '%s' namespace: %w", render.ManagerInternalTLSSecretName, common.OperatorNamespace(), err)
+		}
+
+		// Watch the internal manager TLS secret in the calico namespace, where it's copied for kube-controllers.
+		if err = utils.AddSecretsWatch(c, monitor.PrometheusTLSSecretName, common.TigeraPrometheusNamespace); err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch secret '%s' in '%s' namespace: %w", render.ManagerInternalTLSSecretName, common.OperatorNamespace(), err)
+		}
+
+		// watch for change to primary resource LogCollector
+		err = c.Watch(&source.Kind{Type: &operator.LogCollector{}}, &handler.EnqueueRequestForObject{})
+		if err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch primary resource: %v", err)
+		}
+
+		if ri.manageCRDs {
```

Comment:
I don't think the windows controller needs to add CRD watches either - the CRD management is already handled in the core controller.